db/migrate/20210620144826_add_purpose_to_notifications.rb

```diff
@@ -0,0 +1,16 @@
+class AddPurposeToNotifications < ActiveRecord::Migration[5.2]
+  def up
+    add_column :notifications, :purpose, :string
+    Notification.where(subject: nil).find_each do |notification|
+      notification.update_column(:purpose, "covid_medication_reminder")
+    end
+    Notification.where.not(subject: nil).find_each do |notification|
+      notification.update_column(:purpose, "missed_visit_reminder")
+    end
```

Comment:
(blocking) You can use [update_all](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/update_all) to do all these updates in two SQL statements.  No reason to iterate over every record, especially in a migration where we want things to complete quickly.

Reply:
Does that not lock the db? I thought that was the purpose in batching.

Reply:
We have ~200k notifications? I think it's preferable to lock and update at once than running 200k queries. A good middle ground would be to grab the records in batches of, say, 10,000, instead of one by one. 

Rough pseudocode: `Thing.where(condition).find_in_batches(10000) { |batch| batch.update_all }`

Reply:
Yeah, that seems reasonable. `find_each` uses `find_in_batches(1_000)`, so the only difference is batch size. I think the criticism Rob was initially making is that I should've been updating in batches rather than individually and I misread him as, "don't use find_each". I think using find_each with update_all satisfies his original request.